### ISSUE TITLE
Fail to convert to mitsuba object

### ISSUE THREAD

**Issue by SamTov** [isl-org/Open3D#6416]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

I want to convert my `TraingleMesh` objects to mitsuba objects in order to render them nicely but I get the error:

```bash
mitsuba_object = sphere.to_mitsuba("sphere")
AttributeError: 'open3d.cpu.pybind.geometry.TriangleMesh' object has no attribute 'to_mitsuba'
```

I have confirmed I am using `open3d==0.17.0`

### Steps to reproduce the bug

```python
import open3d as o3d
import mitsuba as mi
import matplotlib.pyplot as plt

print(o3d.__version__)
sphere = o3d.geometry.TriangleMesh.create_sphere(radius=1.0)
sphere.compute_vertex_normals()

mitsuba_object = sphere.to_mitsuba("sphere")

img = mi.render(mitsuba_object)
plt.imshow(img)
plt.show()
```

### Error message

```sh
0.17.0
Traceback (most recent call last):
  File "/Users/samueltovey/work/Repositories/ZINCWARECODE/ZnVis/examples/rendering.py", line 9, in <module>
    mitsuba_object = sphere.to_mitsuba("sphere")
Attrib
uteError: 'open3d.cpu.pybind.geometry.TriangleMesh' object has no attribute 'to_mitsuba'
```

### Expected behavior

I would like a mitsuba object back.

### Open3D, Python and System information

```markdown
- Operating system: macOS Ventura 13.4.1
- Python version: Python 3.10.8
- Open3D version: 0.17.0
- System architecture: apple-silicon m1
- Is this a remote workstation?: no
- How did you install Open3D?: pip
```

**Comment by rsepulveda-biocorellc** [comment]
The `to_mitsuba` is available only for the tensor-based geometries. Try,
```python
sphere = o3d.t.geometry.TriangleMesh.create_sphere(radius=1.0)
sphere.compute_vertex_normals()

mitsuba_object = sphere.to_mitsuba("sphere")
```

Alternatively, if you are working with the legacy geometry classes you can convert to tensor-based geometry prior to creating the mitsuba object as follows
```python
tsphere = o3d.t.geometry.TriangleMesh.from_legacy(sphere)
mitsuba_object = tsphere.to_mitsuba("sphere")

```

**Comment by errissa** [comment]
The `to_mitsuba` function is only available for tensor-based triangle mesh. Try,
```python
sphere = o3d.t.geometry.TriangleMesh.create_sphere(radius=1.0)
sphere.compute_vertex_normals()

mitsuba_object = sphere.to_mitsuba("sphere")
```

Alternatively, you can convert from legacy geometry to tensor-based geometry prior to create the mitsuba object as follows
```python
tsphere = o3d.t.geometry.TriangleMesh.from_legacy(sphere)
mitsuba_object = tsphere.to_mitsuba("sphere")
```

**Comment by SamTov** [comment]
Hey, thanks for the quick response. I tried both methods on an M1 and an intel mac and both failed. For the intel I get the same error as above but now saying:

```sh
mitsuba_object = tsphere.to_mitsuba("sphere")
AttributeError: 'open3d.cpu.pybind.t.geometry.TriangleMesh' object has no attribute 'to_mitsuba'
```
On the M1 I get
```sh
libc++abi: terminating due to uncaught exception of type pybind1::key_error
```

**Comment by errissa** [comment]
You're certain you have Open3d 0.17 installed? I just tested on my Apple Silicon Mac and it worked fine. In fact, I use `to_mitsuba` regularly.

Please check if you can run [this]( https://github.com/isl-org/Open3D/blob/master/examples/python/visualization/to_mitsuba.py) example script.

You might also try the development wheels [here](http://www.open3d.org/docs/latest/getting_started.html). There are some build fixes for Apple Silicon related to exception handling.

**Comment by SamTov** [comment]
Hi, so I am definitely using `0.17.0`. I print it at the start of the script and the output is always the same. I ran the example script in a fresh conda environment just to make sure nothing interfered and I get the following output:

```sh
jit_llvm_init(): detected LLVM version lacks critical functionality, shutting down LLVM backend..
Render mesh with material converted to Mitsuba principled BSDF
libc++abi: terminating due to uncaught exception of type pybind11::stop_iteration: 
```
I include the llvm error here. I am using the `scalar_rgb` backend of mitsuba so I am not sure why this warning is thrown. From the print statements in the example script and output, this fails in the following lines:

```python
print('Render mesh with material converted to Mitsuba principled BSDF')
mi_mesh = mesh.to_mitsuba('monkey')
img = render_mesh(mi_mesh, mesh_center.numpy())
mi.Bitmap(img).write('test.exr')

print('Render mesh with normal-mapped prnincipled BSDF')
```

I ran it again on my script where I do not have a call to `Bitmap` and it fails on the:
```python
mesh.to_mitsuba('monkey')
```
line.

**Comment by errissa** [comment]
I could be wrong but I thought LLVM was required for Mitsuba even if using the `scalar_rgb` variant. From the error you don't have LLVM >= 11.1 installed. Try pip installing the appropriate `1fix6008` wheel from the release [page](https://github.com/isl-org/Open3D/releases). If nothing else, I think it fixes the uncaught exception problem!

**Comment by SamTov** [comment]
The LLVM error is gone. I had to update LLVM on my mac. But I still get the error:
```sh
libc++abi: terminating due to uncaught exception of type pybind11::key_error:
```
when I try to convert to mitsuba.

**Comment by SamTov** [comment]
I managed to get it working. I tried again to use a fresh environment and then install directly from the wheels in the release. This, along with fixing the llvm version, made it work. I don't know if there is a different between the pip target and the wheel because I definitely had the `0.17.0` showing after my `o3d` version print which looks the same now after an install directly from wheels. 

In any case, thank you for all of the help!
